app/controllers/api/custom_buttons_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Api
+  class CustomButtonsController < BaseController
+    def create_resource(_type, _id, data)
+      klass = collection_class(:custom_buttons)
+      data['resource_action'] = add_resource_action_resource(data['resource_action']) if data.key?('resource_action')
```

Comment:
Rather than mutate the data that's passed into this method (which can lead to surprises for the caller). what do you think about creating a new object:

```ruby
custom_button = CustomButton.new(data.except("resource_action"))
custom_button.userid = User.current_user.userid
custom_button.resource_action = add_resource_action_resource(data["resource_action"]) if data.key?("resource_action")
custom_button.save!
```

Reply:
Addressed in 1d85410.
This change required a spec alteration as well.